submissions/LuckyDnepr/task-js-oop/app.js

```diff
@@ -0,0 +1,173 @@
+"use strict";
+
+const blockWidth = 101, //gamefield sizes
+    blockHeight = 83,
+    nBlocksWidth = 5,
+    nBlocksHeight = 6,
+    fieldWidth = blockWidth * nBlocksWidth,
+    fieldHeight = blockHeight * nBlocksHeight,
+    enemyTracksHeight = [62, 145, 228], //coordinates of enemy tracks
+    playerStartPoint = [200, 405],
+    hardestLevel = 4;
+
+let winPoints = 0,
+failPoints = 0;
+
+class Unit {
+    constructor (coordX, coordY, avatar) {
+        this.x = coordX,
+        this.y = coordY;
+        this.sprite = avatar;
+    }
+    render() {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+    update() {}
+    setAvatar (avatar) {
+        this.sprite = avatar;
+    }
+}
+
+class Enemy extends Unit {
+    constructor (coordX, coordY, startSpeed, avatar) {
+        super(coordX, coordY, avatar);
+        this.speed = startSpeed;
+    }
+    update(dt) {
+        this.x += this.speed * dt;
+        this.moveEnemyToStart ();
+        this.checkCollisions();
+    }
+    checkCollisions () {
+        if (this.x > player.x - blockWidth + 20 && //+20 - correction for avatar, for clearer collision
+            this.x < player.x + blockWidth - 15 && //-15 - correction for avatar, for clearer collision
+            this.y < player.y &&
+            this.y + blockHeight > player.y) {
+            endGame(false); //false - you lose the game
+        }
+    }
+    moveEnemyToStart () {
+        if (this.x > fieldWidth + blockWidth) {
+            this.x = Math.random() * -300; //change enemy start point before new appearance
+            this.speed = Math.random() * 150 + Math.random() * 150; //change enemy speed before new appearance
+        }
+    }
+}
+
+class Player extends Unit {
+    constructor (coordX, coordY, avatar) {
+        super(coordX, coordY, avatar);
+        this._freeze = true; //the player does not respond to the keyboard if _freeze === true
```

Comment:
If you feel unhappy with naming, rename rather than add a comment. As is variable contains boolean it most likely reflects some state. E.g. the player `isFrozen`, which reads very close to plain English.